src/charm.py

```diff
@@ -474,6 +480,24 @@ def _on_image_relation_changed(self, _: ops.RelationChangedEvent) -> None:
         self._manager_client.flush_runner()
         self.unit.status = ActiveStatus()
 
+    @catch_charm_errors
+    def _on_database_created(self, _: ops.RelationEvent) -> None:
+        """Handle the MongoDB database created event."""
+        state = self._setup_state()
+        self._setup_service(state)
+
+    @catch_charm_errors
+    def _on_endpoints_changed(self, _: ops.RelationEvent) -> None:
+        """Handle the MongoDB endpoints changed event."""
+        state = self._setup_state()
+        self._setup_service(state)
+
+    @catch_charm_errors
+    def _on_mongodb_relation_broken(self, _: ops.RelationDepartedEvent) -> None:
+        """Handle the MongoDB relation broken event."""
+        state = self._setup_state()
+        self._setup_service(state)
+
```

Comment:
necessary, otherwise the app will not be reconfigured on mongodb changes. especially important when we want to remove reactive mode.